File: backend/go/util/resources.go
package util

import (
	"os"
	"path/filepath"

	"gopkg.in/yaml.v3"
)

type NyTimesConfig struct {
	Nytimes struct {
		ApiKey string `yaml:"apiKey"`
		Uri    string `yaml:"uri"`
	} `yaml:"nytimes"`
}

type OpenWeatherConfig struct {
	OpenWeather struct {
		ApiKey string `yaml:"apiKey"`
		Uri    string `yaml:"uri"`
	} `yaml:"open-weather"`
}

func getApplicationYmlPath() string {
	cwd, err := os.Getwd()
	if err != nil {
		panic("could not get the current working directory")
	}
	path := filepath.Join(cwd, "resources/application.yml")
	return path
}

func GetOpenWeatherConfig() *OpenWeatherConfig {
	buffer, err := os.ReadFile(getApplicationYmlPath())
	if err != nil {
		panic("could not read the application.yml file")
	}

	openWeatherConfig := &OpenWeatherConfig{}
	err = yaml.Unmarshal(buffer, openWeatherConfig)
	if err != nil {
		panic("could not read the application.yml file")
	}
	return openWeatherConfig
}

func GetNyTimesConfig() *NyTimesConfig {
	buffer, err := os.ReadFile(getApplicationYmlPath())
	if err != nil {
		panic("could not read the application.yml file")
	}

	nyTimesConfig := &NyTimesConfig{}
	err = yaml.Unmarshal(buffer, nyTimesConfig)
	if err != nil {
		panic("could not read the application.yml file")
	}
	return nyTimesConfig
}

File: backend/python/util/resources.py
import yaml
from typing import Dict
import os

# Get the current file's directory
base_dir = os.path.dirname(os.path.abspath(__file__))  # Directory of the current file
resources_yaml_path = os.path.join(base_dir, '../resources/application.yml')

class APIResource:
    def __init__(self, api_key: str, uri: str) -> None:
        self.api_key = api_key
        self.uri = uri

def load_config() -> Dict:
    """Load the YAML configuration file."""
    with open(resources_yaml_path, 'r') as file:
        return yaml.safe_load(file)

def get_ny_times_api_resource() -> APIResource:
    config = load_config()
    api_key = config['nytimes']['apiKey']
    uri = config['nytimes']['uri']
    return APIResource(api_key, uri)

def get_simple_weather_api_resource() -> APIResource:
    config = load_config()
    api_key = config['open-weather']['apiKey']
    uri = config['open-weather']['uri']
    return APIResource(api_key, uri)